src/celeritas/ext/GeantImporter.cc

```diff
@@ -239,33 +241,75 @@ import_particles(GeantImporter::DataSelection::Flags particle_flags)
 
 //---------------------------------------------------------------------------//
 /*!
- * Return a populated \c ImportElement vector.
+ * Return a populated pair of \c ImportElement and \c ImportIsotope vectors.
  */
-std::vector<ImportElement> import_elements()
+auto import_elements()
+    -> std::pair<std::vector<ImportElement>, std::vector<ImportIsotope>>
 {
-    auto const& g4element_table = *G4Element::GetElementTable();
-
     std::vector<ImportElement> elements;
+    std::vector<ImportIsotope> isotopes;
+
+    auto const& g4element_table = *G4Element::GetElementTable();
     elements.resize(g4element_table.size());
 
     // Loop over element data
     for (auto const& g4element : g4element_table)
     {
         CELER_ASSERT(g4element);
+        auto const& g4isotope_vec = *g4element->GetIsotopeVector();
+        CELER_ASSERT(g4isotope_vec.size() == g4element->GetNumberOfIsotopes());
 
-        // Add element to vector
+        // Add element to ImportElement vector
         ImportElement element;
         element.name = g4element->GetName();
         element.atomic_number = g4element->GetZ();
         element.atomic_mass = g4element->GetAtomicMassAmu();
         element.radiation_length_tsai = g4element->GetfRadTsai() / (g / cm2);
         element.coulomb_factor = g4element->GetfCoulomb();
 
+        // Despite the function name, this is *NOT* a vector, it's an array
+        double* const g4rel_abundance = g4element->GetRelativeAbundanceVector();
+
+        double total_el_abundance_fraction = 0;  // Verify that the sum is ~1
+        for (auto idx : range(g4element->GetNumberOfIsotopes()))
+        {
+            ImportElement::IsotopeFrac key;
+            key.first = g4isotope_vec[idx]->GetIndex();
+            key.second = g4rel_abundance[idx];
+            element.isotopes_fractions.push_back(std::move(key));
+
+            total_el_abundance_fraction += g4rel_abundance[idx];
+        }
+        CELER_VALIDATE(soft_equal(1., total_el_abundance_fraction),
+                       << "Total relative isotopic abundance for element `"
+                       << element.name
+                       << "` should sum to 1, but instead sum to "
+                       << total_el_abundance_fraction);
+
+        // Populate vector<ImportIsotope>
+        for (auto idx : range(g4isotope_vec.size()))
+        {
+            auto const& g4isotope = *g4isotope_vec.at(idx);
+            CELER_ASSERT(isotopes.size() == g4isotope.GetIndex());
```

Comment:
This definitely won't work if the same isotope is reused in multiple elements. Have you tried loading CMS with debug checking enabled?

Reply:
Good point, I did not. However, the updated code with a separate `import_isotopes()` function works fine with `cms-hllhc-sd.gdml`, so we should be good.